src/detect-flowbits.c

```diff
@@ -69,6 +72,39 @@ void DetectFlowbitsRegister (void)
     DetectSetupParseRegexes(PARSE_REGEX, &parse_regex);
 }
 
+static int FlowbitOrAddData(char *arrptr, DetectFlowbitsData *cd, DetectEngineCtx *de_ctx)
+{
+    char *strarr[MAX_TOKENS];
+    char *token;
+    char *saveptr = NULL;
+    uint8_t i = 0;
+
+    if (strchr(arrptr, ' ') != NULL) {
+        SCLogError(SC_ERR_INVALID_SIGNATURE, "Spaces are not allowed in flowbit names.");
+        return -1;
+    }
+    while ((token = strtok_r(arrptr, "|", &saveptr))) {
+        if (i == MAX_TOKENS) {
+            SCLogError(SC_ERR_INVALID_SIGNATURE, "Number of flowbits exceeds "
+                       "maximum allowed: %d.", MAX_TOKENS);
+            return -1;
+        }
+        strarr[i] = token;
+        arrptr = NULL;
+        i++;
+    }
+
+    cd->or_list = SCMalloc(i * sizeof(uint32_t));
```

Comment:
suggest first setting `cd->or_list_size = i` then passing `cd->or_list_size` to SCMalloc. Result should be the same, but I prefer making it very clear how size vars map to allocations.

Reply:
another suggestion: `SCCalloc(cd->or_list_size, sizeof(uint32_t));`